src/server/json_family.cc

```diff
@@ -157,6 +160,128 @@ OpResult<json> GetJson(const OpArgs& op_args, string_view key) {
   return decoder.get_result();
 }
 
+// Returns the index of the next right bracket
+optional<size_t> GetNextIndex(string_view str) {
+  size_t end_val_idx = string::npos;
+  size_t current_idx = 0;
+  while (current_idx + 1 < str.size()) {
+    if (end_val_idx != string::npos) {
+      break;
+    }
+
+    // ignore escaped character after the backslash (e.g. \').
+    if (str[current_idx] == '\\') {
+      current_idx += 2;
+    } else if (str[current_idx] == '\'' && str[current_idx + 1] == ']') {
+      end_val_idx = current_idx++;
```

Comment:
why not just return here? 
seems like something that will remove other checks as well.